rpc/src/v1/types/filter.rs

```diff
@@ -70,17 +75,29 @@ pub struct Filter {
 	pub limit: Option<usize>,
 }
 
-impl Into<EthFilter> for Filter {
-	fn into(self) -> EthFilter {
+impl Filter {
+	pub fn try_into(self) -> Result<EthFilter, RpcError> {
+		if self.block_hash.is_some() && (self.from_block.is_some() || self.to_block.is_some()) {
+			return Err(invalid_params("blockHash", "blockHash is mutually exclusive with fromBlock/toBlock"));
+		}
+
 		let num_to_id = |num| match num {
 			BlockNumber::Num(n) => BlockId::Number(n),
 			BlockNumber::Earliest => BlockId::Earliest,
 			BlockNumber::Latest | BlockNumber::Pending => BlockId::Latest,
 		};
 
-		EthFilter {
-			from_block: self.from_block.map_or_else(|| BlockId::Latest, &num_to_id),
-			to_block: self.to_block.map_or_else(|| BlockId::Latest, &num_to_id),
+		let (from_block, to_block) = match self.block_hash {
+			Some(hash) => {
+				let hash = hash.into();
+				(BlockId::Hash(hash), BlockId::Hash(hash))
+			},
+			None => (self.from_block.map_or_else(|| BlockId::Latest, &num_to_id),
+					 self.to_block.map_or_else(|| BlockId::Latest, &num_to_id)),
```

Comment:
This indentation displays correctly on my editor. I think why it looks weird here is probably because the diff's `+` mark pushed tabs one level deeper.

Reply:
Let me wrap that to a new line anyway.